docset/winserver2025-ps/smbshare/Set-SmbServerConfiguration.md

```diff
@@ -851,6 +977,54 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -Smb2DialectMax
+
+This parameter specifies the maximum version of the SMB protocol to be used. Acceptable values are:
+
+- None – There is no maximum protocol version specified, the server can use any supported version.
+- SMB202 – SMB 2.0.2 is the maximum version accepted by the SMB Sever
+- SMB210 - SMB 2.1.0 is the maximum version accepted by the SMB Sever
+- SMB300 - SMB 3.0.0 is the maximum version accepted by the SMB Sever
+- SMB302 - SMB 3.0.2 is the maximum version accepted by the SMB Sever
+- SMB311 - SMB 3.1.1 is the maximum version accepted by the SMB Sever
+
+```yaml
+Type: SwitchParameter
+Parameter Sets: (All)
+Aliases:
+Accepted values: None, SMB202, SMB210, SMB300, SMB302, SMB311
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Smb2DialectMin
+
+This parameter specifies the minimum version of the SMB protocol to be used. Acceptable values are:
+
+- None – There is no maximum protocol version specified, the server can use any supported version.
+- SMB202 – SMB 2.0.2 is the maximum version accepted by the SMB Sever
+- SMB210 - SMB 2.1.0 is the maximum version accepted by the SMB Sever
+- SMB300 - SMB 3.0.0 is the maximum version accepted by the SMB Sever
+- SMB302 - SMB 3.0.2 is the maximum version accepted by the SMB Sever
+- SMB311 - SMB 3.1.1 is the maximum version accepted by the SMB Sever
+
+```yaml
+Type: Smb2DialectMin
+Parameter Sets: (All)
+Aliases:
+Accepted values: None, SMB202, SMB210, SMB300, SMB302, SMB311
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
 ### -SmbServerNameHardeningLevel
```

Comment:
Controls the level of validation that a server performs on the service principal name (SPN) that is provided by the client device when the client establishes a session using Server Message Block (SMB). The possible values are 0-2.

0: Don't enforce SPN check
1: Allow clients who did not provide the target, but fail those who do provide the target and it doesn't match
2: Only allow clients who supply matching targets

(This should be fixed in downlevel versions of powershell too, it was never good.)

Reply:
@Xelu86, apply this change to WS22:

### -SmbServerNameHardeningLevel

Controls the level of validation that a server performs on the service principal name (SPN) that is
provided by the client device when the client establishes a session using Server Message Block
(SMB). The acceptable values are:

- `0`: Don't enforce SPN check.
- `1`: Allow clients who didn't provide the target, but fail those who do provide the target and it
  doesn't match.
- `2`: Only allow clients who supply matching targets.

```yaml
Type: UInt32
Parameter Sets: (All)
Aliases:
Accepted values: 0, 1, 2

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```